arch/riscv/core/thread.c

```diff
@@ -269,14 +274,17 @@ FUNC_NORETURN void arch_user_mode_enter(k_thread_entry_t user_entry,
 	register void *a1 __asm__("a1") = p1;
 	register void *a2 __asm__("a2") = p2;
 	register void *a3 __asm__("a3") = p3;
-#if defined(CONFIG_THREAD_LOCAL_STORAGE
-	register void *tp __asm__("tp") = _current->tls;
+#if defined(CONFIG_THREAD_LOCAL_STORAGE)
+	register void *tp __asm__("tp") = (void *) _current->tls;
```

Comment:
Need to merge these 2 lines to the previous commit to avoid breaking git bisect.

Reply:
This hunk has been removed in the most recent update.